tests/scripts/all.sh

```diff
@@ -1274,6 +1274,38 @@ component_test_crypto_full_no_cipher () {
     make test
 }
 
+component_test_crypto_default_stream_cipher_only () {
+    msg "build: default with only stream cipher"
+
+    # Disable all ciphers
+    # AEAD (controlled by the presence of one of GCM_C, CCM_C, CHACHAPOLY_C
+    scripts/config.py unset MBEDTLS_GCM_C
+    scripts/config.py unset MBEDTLS_CCM_C
+    scripts/config.py unset MBEDTLS_CHACHAPOLY_C
+    # CBC-legacy (controlled by MBEDTLS_CIPHER_MODE_CBC plus at least one block cipher (AES, ARIA, Camellia, DES))
+    scripts/config.py unset MBEDTLS_CIPHER_MODE_CBC
+    scripts/config.py unset MBEDTLS_AES_C
+    scripts/config.py unset MBEDTLS_CAMELLIA_C
+    scripts/config.py unset MBEDTLS_ARIA_C
+    scripts/config.py unset MBEDTLS_DES_C
+    # CBC-EtM (controlled by the same as CBC-legacy plus MBEDTLS_SSL_ENCRYPT_THEN_MAC)
+    scripts/config.py unset MBEDTLS_SSL_ENCRYPT_THEN_MAC
+    # stream (currently that's just the NULL pseudo-cipher (controlled by MBEDTLS_CIPHER_NULL_CIPHER)
+    scripts/config.py unset MBEDTLS_CIPHER_NULL_CIPHER
+    # Indirect dependencies
+    scripts/config.py unset MBEDTLS_CTR_DRBG_C
+    scripts/config.py unset MBEDTLS_CMAC_C
+    scripts/config.py unset MBEDTLS_NIST_KW_C
+
+    # Enable stream(null) cipher only
+    scripts/config.py set MBEDTLS_CIPHER_NULL_CIPHER
+    make
+
+    msg "test: default with only stream cipher"
+    make test
```

Comment:
I tried to run `ssl-opt.sh` with `TLS 1.2` filter to limit executed tests and it works for `CBC` and `CBC-EtM`, but for `Stream/NULL` not and I wonder if this make sense for Stream/NULL:

```
******************************************************************
* test_tls1_2_default_stream_cipher_only: test: default with only stream cipher - ssl-opt.sh (subset) 
* czw, 29 wrz 2022, 13:53:24 CEST
******************************************************************
CBC Record splitting: TLS 1.2, no splitting ............................ SKIP
Certificate hash: client TLS 1.2 -> SHA-2 .............................. PASS
Small client packet TLS 1.2 BlockCipher ................................ SKIP
Small client packet TLS 1.2 BlockCipher, without EtM ................... SKIP
Small client packet TLS 1.2 BlockCipher larger MAC ..................... SKIP
Small client packet TLS 1.2 AEAD ....................................... SKIP
Small client packet TLS 1.2 AEAD shorter tag ........................... SKIP
Small client packet DTLS 1.2 ........................................... SKIP
Small client packet DTLS 1.2, without EtM .............................. SKIP
Small server packet TLS 1.2 BlockCipher ................................ SKIP
Small server packet TLS 1.2 BlockCipher, without EtM ................... SKIP
Small server packet TLS 1.2 BlockCipher larger MAC ..................... SKIP
Small server packet TLS 1.2 AEAD ....................................... SKIP
Small server packet TLS 1.2 AEAD shorter tag ........................... SKIP
Small server packet DTLS 1.2 ........................................... SKIP
Small server packet DTLS 1.2, without EtM .............................. SKIP
Large client packet TLS 1.2 BlockCipher ................................ SKIP
Large client packet TLS 1.2 BlockCipher, without EtM ................... SKIP
Large client packet TLS 1.2 BlockCipher larger MAC ..................... SKIP
Large client packet TLS 1.2 AEAD ....................................... SKIP
Large client packet TLS 1.2 AEAD shorter tag ........................... SKIP
Large server packet TLS 1.2 BlockCipher ................................ SKIP
Large server packet TLS 1.2 BlockCipher, without EtM ................... SKIP
Large server packet TLS 1.2 BlockCipher larger MAC ..................... SKIP
Large server packet TLS 1.2 BlockCipher, without EtM, truncated MAC .... SKIP
Large server packet TLS 1.2 AEAD ....................................... SKIP
Large server packet TLS 1.2 AEAD shorter tag ........................... SKIP
DTLS fragmenting: gnutls server, DTLS 1.2 .............................. RETRY(client-timeout) RETRY(client-timeout) FAIL
  ! bad client exit code (expected 0, got 143)
  ! outputs saved to o-XXX-28.log
DTLS fragmenting: gnutls client, DTLS 1.2 .............................. FAIL
  ! bad client exit code (expected 0, got 1)
  ! outputs saved to o-XXX-29.log
DTLS fragmenting: openssl server, DTLS 1.2 ............................. FAIL
  ! bad client exit code (expected 0, got 1)
  ! outputs saved to o-XXX-30.log
DTLS fragmenting: openssl client, DTLS 1.2 ............................. FAIL
  ! bad client exit code (expected 0, got 1)
  ! outputs saved to o-XXX-31.log
DTLS fragmenting: 3d, gnutls server, DTLS 1.2 .......................... SKIP
DTLS fragmenting: 3d, gnutls client, DTLS 1.2 .......................... SKIP
DTLS fragmenting: 3d, openssl server, DTLS 1.2 ......................... SKIP
DTLS fragmenting: 3d, openssl client, DTLS 1.2 ......................... SKIP
TLS 1.3: Not supported version check:gnutls: srv max TLS 1.2 ........... SKIP
TLS 1.3: Not supported version check:openssl: srv max TLS 1.2 .......... SKIP
TLS 1.2: Check rsa_pss_rsae compatibility issue, m->O .................. SKIP
TLS 1.2: Check rsa_pss_rsae compatibility issue, m->G .................. SKIP
------------------------------------------------------------------------
FAILED (35 / 39 tests (34 skipped))

```

Most are skipped and fails are caused because cipher-suits do not match:
```
Error in handshake(): No supported cipher suites have been found.
Error in handshake(): No supported cipher suites have been found.
Error in handshake(): No supported cipher suites have been found.
Error in handshake(): No supported cipher suites have been found.
Error in handshake(): No supported cipher suites have been found.
Exiting via signal 15
-------
ssl_tls12_server.c:1683: |1| got no ciphersuites in common
-------
ssl_tls12_server.c:1683: |1| got no ciphersuites in common
-------
140021506344256:error:1417A0C1:SSL routines:tls_post_process_client_hello:no shared cipher:../ssl/statem/statem_srvr.c:2283:
```

Reply:
Ah, I assume gnutls and openssl don't offer/accept NULL ciphersuites by default (which is of course extremely sensible as these ciphersuite do not provide any confidentiality), so evidently trying to interop with them in a build where we only support NULL ciphersuite is not going to work well.

I'm not sure how to properly express that in terms of `requires_config_enabled` dependencies... Perhaps the simplest thing to do is to just invoke `ssl-opt.sh` with `-e gnutls|openssl` in this component?

Reply:
Hmm, I don't understand, the CI seems to be passing now. What changed?

Reply:
Oh I see now, `ssl-opt.sh` was added to the other components but not this one. My suggestion stands, then (with a comment explaining why). (Also, perhaps it's `-e 'gnutls\|openssl` actually. And perhaps we need to also exclude things like `m->O`, `G->m` etc.)

Reply:
Yes, I removed this because even if we skip `gnutls\|openssl` (4 failures) then only one test will be executed: `Certificate hash: client TLS 1.2 -> SHA-2`.

Reply:
Ah, good point. Probably best to leave a comment then so that we don't wonder about it in the future, for example "Not running ssl-opt.sh because most tests require a non-NULL ciphersuite."

Reply:
`ssl-opt.sh` was originally written with a “normal” configuration in mind, using certificates. Until relatively recently, it couldn't do anything useful in a PSK-only build. Then I added instrumentation to adapt the command lines of `CLI_CMD` and `SRV_CMD` to pass `-psk …` or similar so that test cases that aren't specifically related to asymmetric cryptography can run. For null ciphersuites, we'd need similar instrumentation to add some form of `-allow-null-ciphersuites`. I don't know if it's worth the effort.

Reply:
Good point, Gilles, but I don't think it's worth the effort for now. NULL ciphersuites are quite niche, so I don't think we need to pro-actively improve our support/testing for them; at least that was not the goal of this issue, but rather to catch an fix mistakes in `#ifdef`s that would have been a distraction when working on builds with driver-only ciphers/AEAD, ie start this work from a clean(er) state.